src/rtctools_heat_network/workflows/simulator_workflow.py

```diff
@@ -324,55 +326,73 @@ def times(self, variable=None) -> np.ndarray:
 
 
 class NetworkSimulatorHIGHSWeeklyTimeStep(NetworkSimulatorHIGHS):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.__indx_max_peak = None
+        self.__day_steps = 5
+
+    def parameters(self, ensemble_member):
+        parameters = super().parameters(ensemble_member)
+        parameters["peak_day_index"] = self.__indx_max_peak
+        parameters["time_step_days"] = self.__day_steps
+        return parameters
+
     def read(self):
         """
-        Reads the yearly profile with hourly time steps and adapt to a daily averaged profile.
+        Reads the yearly profile with hourly time steps and adapt to a daily averaged profile
+        except for the day with the peak demand.
         """
-        # TODO: moves functions like these to a local place for repatitive use. Reduce code
-        # duplication
         super().read()
 
-        demands = self.heat_network_components.get("demand", [])
-        new_datastore = DataStore(self)
-        new_datastore.reference_datetime = self.io.datetimes[0]
-
-        for ensemble_member in range(self.ensemble_size):
-            total_demand = sum(
-                self.get_timeseries(f"{demand}.target_heat_demand", ensemble_member).values
-                for demand in demands
+        self.__indx_max_peak, _ = adapt_hourly_profile_to_common(self, self.__day_steps)
+        logger.info("HeatProblem read")
+
+    def _set_data_with_averages_and_peak_day(
+        self,
+        datastore: DataStore,
+        variable_name: str,
+        ensemble_member: int,
+        new_date_times: np.array,
+    ):
```

Comment:
I would expect this function to also be in the adapt_profiles.py, as it is generic for both workflows. Was there a reason not to do that?

Reply:
My mistake

Reply:
Fixed